crates/sequencer/src/state.rs

```diff
@@ -16,9 +22,235 @@ use starknet_api::{
 /// See [rustc-hash](https://crates.io/crates/rustc-hash) for more information.
 #[derive(Default)]
 pub struct State {
-    pub classes: FxHashMap<ClassHash, ContractClass>,
-    pub compiled_classes: FxHashMap<ClassHash, CompiledClassHash>,
-    pub contracts: FxHashMap<ContractAddress, ClassHash>,
-    pub storage: FxHashMap<ContractStorageKey, StarkFelt>,
-    pub nonces: FxHashMap<ContractAddress, Nonce>,
+    classes: FxHashMap<ClassHash, ContractClass>,
+    compiled_class_hashes: FxHashMap<ClassHash, CompiledClassHash>,
+    contracts: FxHashMap<ContractAddress, ClassHash>,
+    storage: FxHashMap<ContractStorageKey, StarkFelt>,
+    nonces: FxHashMap<ContractAddress, Nonce>,
+}
+
+/// State implementation for the sequencer. We use a mutable reference to the state
+/// because this is what will be available during the implementation of the execution.
+impl<'a> BlockifierState for &'a mut State {
+    fn set_storage_at(
+        &mut self,
+        contract_address: ContractAddress,
+        key: StorageKey,
+        value: StarkFelt,
+    ) {
+        self.storage.insert((contract_address, key), value);
+    }
+
+    fn increment_nonce(&mut self, contract_address: ContractAddress) -> StateResult<()> {
+        let current_nonce = self
+            .nonces
+            .get(&contract_address)
+            .cloned()
+            .unwrap_or_default();
+
+        let mut current_nonce: u64 = current_nonce.0.try_into()?;
+        current_nonce += 1;
+
+        self.nonces
+            .insert(contract_address, Nonce(StarkFelt::from(current_nonce)));
+
+        Ok(())
+    }
+
+    fn set_class_hash_at(
+        &mut self,
+        contract_address: ContractAddress,
+        class_hash: ClassHash,
+    ) -> StateResult<()> {
+        self.contracts.insert(contract_address, class_hash);
+        Ok(())
+    }
+
+    fn set_contract_class(
+        &mut self,
+        class_hash: &ClassHash,
+        contract_class: ContractClass,
+    ) -> StateResult<()> {
+        self.classes.insert(class_hash.to_owned(), contract_class);
+        Ok(())
+    }
+
+    fn set_compiled_class_hash(
+        &mut self,
+        class_hash: ClassHash,
+        compiled_class_hash: CompiledClassHash,
+    ) -> StateResult<()> {
+        self.compiled_class_hashes
+            .insert(class_hash, compiled_class_hash);
+        Ok(())
+    }
+
+    fn to_state_diff(&self) -> CommitmentStateDiff {
+        unreachable!("to_state_diff should not be called in the sequencer")
+    }
+}
+
+impl<'a> BlockifierStateReader for &'a mut State {
+    /// Default: 0 for an uninitialized contract address.
```

Comment:
This comment isn't so clear
default for what? 

you're saying: storage.get(&(contract_address, key)) will fail if contract_address isn't initialized and thus it'll trigger default case of unwrap_or_default?

Reply:
Will add `contract address or key`. It just means that if the storage value in the contract is uninitialzed, we return a default of 0.